### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by Kief5555**
Added error handling

**Please describe the changes this PR makes and why it should be merged:**
Because its missing from the debug option

**Status and versioning classification:**

Added logging to errors and warning when DEBUG mode is enabled.

### CODE REVIEW

File: lib/DiscordMusicBot.js [hunk 1]
```diff
@@ -101,6 +101,8 @@ class DiscordMusicBot extends Client {
       this.warn("Only enable this if you know what you are doing!");
       process.on("unhandledRejection", (error) => console.log(error));
       process.on("uncaughtException", (error) => console.log(error));
+      process.on("warning", (error) => console.log(error));
+			process.on("error", (error) => console.log(error));
```
Comment (SudhanPlayz) [inline lib/DiscordMusicBot.js:105]: There is no process event called `error`

**Review by SudhanPlayz — CHANGES_REQUESTED** [review]

File: lib/DiscordMusicBot.js (hunk 1 above)
Comment (ItzCuteAkemi) [reply lib/DiscordMusicBot.js:105]: Hmm how could this thing will make everything is verbose in console

Comment (ItzCuteAkemi) [reply lib/DiscordMusicBot.js:105]: Ye I agree there's no event called `error`

Comment (Kief5555) [reply lib/DiscordMusicBot.js:105]: Hmm.. It worked for me.

Comment (Kief5555) [reply lib/DiscordMusicBot.js:105]: Ah, it was the unhandled rejection. I guess the warning would be an addition.

**Comment by Kief5555** [comment]
@SudhanPlayz fixed

**Comment by JotaroKujo0525** [comment]
I'm going to test this

**Comment by JotaroKujo0525** [comment]
Screenshot_20220809-181419
apparently, this commit still don't solve the issue with the error handling

**Comment by JotaroKujo0525** [comment]
hello! i'm here to inform you that this error is rate limiting, there is no problem where you run the code. basically this is how it worked for you.

you waited long enough and the rate limit has been lifted so that's why you can login again

**Comment by JotaroKujo0525** [comment]
discord.js is not outputting error with this matter, so that's why it's kinda confusing in first nit seeing your bot login, also the
```
process.on("warning")
```

is irrelevant since we're not using the `--no-warnings` and the warnings will be shows as usual

**Comment by JotaroKujo0525** [comment]
so tl;dr
the current debug mode is already fine

**Comment by XstreamSpeed** [comment]
> so tl;dr the current debug mode is already fine, there's some errors you can't see, and i thought this changes it but nah it still doesn't. in my experience with v5, debug mode is outputting the errors i need

To be honest, you could just use a dedicated anticrash file for this which would be much more better for developers actively contributing to this project, tons of anticrash.js files are available on the internet and I've written my own below:

```js
const chalk = require("chalk"); // Importing Chalk from Chalk
const { MessageEmbed, WebhookClient } = require('discord.js') // Importing MessageEmbed from Discord.js
const { inspect } = require("util");
const s = new WebhookClient({
                id: "id",
                token:"token",
            });
//not using normal client.channels.cache.get("id") functions as it returns bunch of errors.

module.exports = (client) => {

client.on('error', err => {
        // const a = client.channels.cache.get(config.ERROR_LOG_CHANNEL)
        console.log(err)
        const ErrorEmbed = new MessageEmbed()
            .setTitle('Error')
            .setURL('https://discordjs.guide/popular-topics/errors.html#api-errors')
            .setColor('#2F3136')
            .setDescription(`\`\`\`${inspect(error, {depth: 0})}\`\`\``)
            .setTimestamp()
        return s.send({
            embeds: [ErrorEmbed]
        })
    });

    process.on("unhandledRejection", (reason, p) => {
        console.log(
            chalk.yellow('[Anticrash] Unhandled Rejection\n'),
            reason, p
        )
        const unhandledRejectionEmbed = new MessageEmbed()
            .setTitle('**[Anticrash] Unhandled Rejection**')
            .setURL('https://nodejs.org/api/process.html#event-unhandledrejection')
            .setColor("#FF5555")
            .addField('Reason', `\`\`\`js\n${inspect(reason, { depth: 0 })}\`\`\``.substring(0, 1000))
            .addField('Promise', `\`\`\`js\n${inspect(p, { depth: 0 })}\`\`\``.substring(0, 1000))
            .setTimestamp()
        return s.send({
            embeds: [unhandledRejectionEmbed]
        })
    });
    
    process.on("uncaughtException", (err, origin) => {
        console.log(err, origin)
        const uncaughtExceptionEmbed = new MessageEmbed()
            .setTitle('**[Anticrash] Uncaught Exception**')
            .setColor("#FF5555")
            .setURL('https://nodejs.org/api/process.html#event-uncaughtexception')
            .addField('Error', `\`\`\`js\n${inspect(err, { depth: 0 })}\`\`\``.substring(0, 1000))
            .addField('Origin', `\`\`\`js\n${inspect(origin, { depth: 0 })}\`\`\``.substring(0, 1000))
            .setTimestamp()
        return s.send({
            embeds: [uncaughtExceptionEmbed]
        })
    });
    
    process.on("uncaughtExceptionMonitor", (err, origin) => {
        console.log(err, origin)
        const xd = new MessageEmbed()
            .setTitle('**[Anticrash] Uncaught Exception Monitor*')
            .setColor("#FF5555")
            .setURL('https://nodejs.org/api/process.html#event-uncaughtexceptionmonitor')
            .addField('Error', `\`\`\`js\n${inspect(err, { depth: 0 })}\`\`\``.substring(0, 1000))
            .addField('Origin', `\`\`\`js\n${inspect(origin, { depth: 0 })}\`\`\``.substring(0, 1000))
            .setTimestamp();
          return s.send({
            embeds: [xd]
        })
    });
    
    process.on("multipleResolves", (type, promise, reason) => {
        console.log(type, promise, reason)
        const multipleResolvesEmbed = new MessageEmbed()
            .setTitle('**[Anticrash] Multiple Resolves**')
            .setURL('https://nodejs.org/api/process.html#event-multipleresolves')
            .setColor("#FF5555")
            .addField('Type', `\`\`\`js\n${inspect(type, { depth: 0 })}\`\`\``.substring(0, 1000))
            .addField('Promise', `\`\`\`js\n${inspect(promise, { depth: 0 })}\`\`\``.substring(0, 1000))
            .addField('Reason', `\`\`\`js\n${inspect(reason, { depth: 0 })}\`\`\``.substring(0, 1000))
            .setTimestamp()
        return s.send({
            embeds: [multipleResolvesEmbed]
        })
    });
}

**Comment by XstreamSpeed** [comment]
> > so tl;dr the current debug mode is already fine, there's some errors you can't see, and i thought this changes it but nah it still doesn't. in my experience with v5, debug mode is outputting the errors i need
> 
> To be honest, you could just use a dedicated anticrash file for this which would be much more better for developers actively contributing to this project, tons of anticrash.js files are available on the internet and I've written my own below:
> 
> ```js
> const chalk = require("chalk"); // Importing Chalk from Chalk
> const { MessageEmbed, WebhookClient } = require('discord.js') // Importing MessageEmbed from Discord.js
> const { inspect } = require("util");
> const s = new WebhookClient({
>                 id: "id",
>                 token:"token",
>             });
> //not using normal client.channels.cache.get("id") functions as it returns bunch of errors.
> 
> module.exports = (client) => {
> 
> client.on('error', err => {
>         // const a = client.channels.cache.get(config.ERROR_LOG_CHANNEL)
>         console.log(err)
>         const ErrorEmbed = new MessageEmbed()
>             .setTitle('Error')
>             .setURL('https://discordjs.guide/popular-topics/errors.html#api-errors')
>             .setColor('#2F3136')
>             .setDescription(`\`\`\`${inspect(error, {depth: 0})}\`\`\``)
>             .setTimestamp()
>         return s.send({
>             embeds: [ErrorEmbed]
>         })
>     });
> 
>     process.on("unhandledRejection", (reason, p) => {
>         console.log(
>             chalk.yellow('[Anticrash] Unhandled Rejection\n'),
>             reason, p
>         )
>         const unhandledRejectionEmbed = new MessageEmbed()
>             .setTitle('**[Anticrash] Unhandled Rejection**')
>             .setURL('https://nodejs.org/api/process.html#event-unhandledrejection')
>             .setColor("#FF5555")
>             .addField('Reason', `\`\`\`js\n${inspect(reason, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .addField('Promise', `\`\`\`js\n${inspect(p, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .setTimestamp()
>         return s.send({
>             embeds: [unhandledRejectionEmbed]
>         })
>     });
>     
>     process.on("uncaughtException", (err, origin) => {
>         console.log(err, origin)
>         const uncaughtExceptionEmbed = new MessageEmbed()
>             .setTitle('**[Anticrash] Uncaught Exception**')
>             .setColor("#FF5555")
>             .setURL('https://nodejs.org/api/process.html#event-uncaughtexception')
>             .addField('Error', `\`\`\`js\n${inspect(err, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .addField('Origin', `\`\`\`js\n${inspect(origin, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .setTimestamp()
>         return s.send({
>             embeds: [uncaughtExceptionEmbed]
>         })
>     });
>     
>     process.on("uncaughtExceptionMonitor", (err, origin) => {
>         console.log(err, origin)
>         const xd = new MessageEmbed()
>             .setTitle('**[Anticrash] Uncaught Exception Monitor*')
>             .setColor("#FF5555")
>             .setURL('https://nodejs.org/api/process.html#event-uncaughtexceptionmonitor')
>             .addField('Error', `\`\`\`js\n${inspect(err, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .addField('Origin', `\`\`\`js\n${inspect(origin, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .setTimestamp();
>           return s.send({
>             embeds: [xd]
>         })
>     });
>     
>     process.on("multipleResolves", (type, promise, reason) => {
>         console.log(type, promise, reason)
>         const multipleResolvesEmbed = new MessageEmbed()
>             .setTitle('**[Anticrash] Multiple Resolves**')
>             .setURL('https://nodejs.org/api/process.html#event-multipleresolves')
>             .setColor("#FF5555")
>             .addField('Type', `\`\`\`js\n${inspect(type, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .addField('Promise', `\`\`\`js\n${inspect(promise, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .addField('Reason', `\`\`\`js\n${inspect(reason, { depth: 0 })}\`\`\``.substring(0, 1000))
>             .setTimestamp()
>         return s.send({
>             embeds: [multipleResolvesEmbed]
>         })
>     });
> }
> ```

PS: You need to add: ```js
require("events/anticrash") in the index.js file!
```

**Comment by Kief5555** [comment]
You know what, you guys keep arguing with me, disrespectful. Im not going to continue this pull request.

**Comment by XstreamSpeed** [comment]
> You know what, you guys keep arguing with me, disrespectful. Im not going to continue this pull request.

.... w h a t

**Comment by JotaroKujo0525** [comment]
tf

**Pull request closed without merging**